cmd/shared.go

```diff
@@ -0,0 +1,40 @@
+package cmd
+
+import (
+	"fmt"
+)
+
+// DatastoreEngine type to define different engine types
+type DatastoreEngine string
+
+// Types of Datastore Engines
+const (
+	Memory   DatastoreEngine = "memory"
+	Postgres DatastoreEngine = "postgres"
+	MySQL    DatastoreEngine = "mysql"
+)
+
+// String converts DatastoreEngine e to string
+func (e DatastoreEngine) String() string {
+	return string(e)
+}
+
+// IsValid checks if DatastoreEngine e is valid
+func (e DatastoreEngine) IsValid() bool {
+	var isValid bool
+	for _, engineType := range []DatastoreEngine{Memory, Postgres, MySQL} {
```

Comment:
This code can change to simply:

```go
for _, engine := range []DatastoreEngine{Memory, Postgres, MySQL} {
    if engine.String() == e.String() {
        return true
    }
}

return false
```

The variable assignment and then return is redundant.